mavros_extras/src/plugins/fake_gps.cpp

```diff
@@ -266,6 +225,9 @@ class FakeGPSPlugin : public plugin::PluginBase,
 		old_ecef = current_ecef;
 
 		UAS_FCU(m_uas)->send_message_ignore_drop(fix);
+
+		// Throttle incoming messages to 5hz
+		gps_rate.sleep();
```

Comment:
No! Sleeping here is not good idea. You may convert Rate to Duration.
Or look how rate works, maybe it has "elapsed" method?